From: Jiri Slaby <jslaby@suse.cz>
Date: Thu, 3 May 2018 15:51:17 +0200
Subject: Revert "mtd: cfi: cmdset_0001: Workaround Micron Erase suspend bug."
Patch-mainline: never, kabi
References: kabi

This reverts commit d519eb768c1e17f1494d539eafdc6c4f0ba318b2, upstream
commit 46a16a2283f9e678a4e26829175e0c37a5191860. It depends on "mtd:
cfi: cmdset_0001: Do not allow read/write to suspend erase block." which
was just reverted.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 drivers/mtd/chips/cfi_cmdset_0001.c | 17 -----------------
 1 file changed, 17 deletions(-)

diff --git a/drivers/mtd/chips/cfi_cmdset_0001.c b/drivers/mtd/chips/cfi_cmdset_0001.c
index 484cc4646521..286b97a304cf 100644
--- a/drivers/mtd/chips/cfi_cmdset_0001.c
+++ b/drivers/mtd/chips/cfi_cmdset_0001.c
@@ -45,7 +45,6 @@
 #define I82802AB	0x00ad
 #define I82802AC	0x00ac
 #define PF38F4476	0x881c
-#define M28F00AP30	0x8963
 /* STMicroelectronics chips */
 #define M50LPW080       0x002F
 #define M50FLW080A	0x0080
@@ -376,17 +375,6 @@ static void cfi_fixup_major_minor(struct cfi_private *cfi,
 		extp->MinorVersion = '1';
 }
 
-static int cfi_is_micron_28F00AP30(struct cfi_private *cfi, struct flchip *chip)
-{
-	/*
-	 * Micron(was Numonyx) 1Gbit bottom boot are buggy w.r.t
-	 * Erase Supend for their small Erase Blocks(0x8000)
-	 */
-	if (cfi->mfr == CFI_MFR_INTEL && cfi->id == M28F00AP30)
-		return 1;
-	return 0;
-}
-
 static inline struct cfi_pri_intelext *
 read_pri_intelext(struct map_info *map, __u16 adr)
 {
@@ -838,11 +826,6 @@ static int chip_ready (struct map_info *map, struct flchip *chip, unsigned long
 			goto sleep;
 
 
-		/* do not suspend small EBs, buggy Micron Chips */
-		if (cfi_is_micron_28F00AP30(cfi, chip) &&
-		    (chip->in_progress_block_mask == ~(0x8000-1)))
-			goto sleep;
-
 		/* Erase suspend */
 		map_write(map, CMD(0xB0), adr);
 
-- 
2.16.3

